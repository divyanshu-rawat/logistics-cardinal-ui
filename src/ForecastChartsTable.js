import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import styled from 'styled-components';
import { TableHeaderColumn } from 'react-bootstrap-table';
import { LineChart, Line, XAxis, Tooltip } from 'recharts';

import Table from 'components/Table';

const TableWithoutOverflow = styled(Table)`
  overflow: inherit !important;
  .react-bs-container-body {
    overflow: inherit !important;
  }
  .table > tbody > tr > td {
    padding: 0;
    outline: 0;
    border-right: 1px solid ${(props) => props.theme.greyLight};
    &:last-child {
      border: 0;
    }
    &:first-child {
      border-right-width: 2px;
    }
    &:focus,
    &:active {
      outline: 0;
    }
  }
  .charts_cell {
    overflow: inherit !important;
    background-image: linear-gradient(
      to top,
      ${(props) => props.theme.greyMedium},
      ${(props) => props.theme.greyDeep}
    );
  }
  .recharts-tooltip-wrapper {
    z-index: 1000;
  }
`;

export class ForecastChartsTable extends Component {
  static propTypes = {
    chartData: PropTypes.object.isRequired,
    sortedDays: PropTypes.array.isRequired,
  };

  state = {
    cellWidth: 0,
  };

  componentDidMount() {
    this.setState({ cellWidth: this.chartDiv.offsetWidth }); // eslint-disable-line react/no-did-mount-set-state
  }

  chartFormatter = (cell, width) => (
    <LineChart width={width} height={width} data={cell.data}>
      <XAxis dataKey="formattedTime" hide />
      <Tooltip wrapperStyle={{ backgroundColor: '#343941', left: '-120px' }} />
      {cell.sortedDates.map((date) => (
        <Line
          key={`${date}${cell.data.formattedTime}`}
          type="natural"
          dot={false}
          dataKey={date}
          stroke={cell.colors[date].lineColor}
        />
      ))}
    </LineChart>
  );

  cellFormatter = (cell) => (
    <div
      ref={(element) => {
        this.chartDiv = element;
      }}
      className="chart-container"
    >
      {this.state.cellWidth && this.chartFormatter(cell, this.state.cellWidth)}
    </div>
  );

  render() {
    const { chartData, sortedDays } = this.props;
    return (
      <TableWithoutOverflow
        data={[chartData]}
        pagination={false}
        tableHeaderClass="hidden"
      >
        <TableHeaderColumn
          width="100px"
          isKey
          dataField="week"
          columnClassName="charts_cell"
          csvHeader="week"
        >
          Week
        </TableHeaderColumn>
        {sortedDays.map((dayNr) => {
          const dayStr = moment()
            .day(dayNr)
            .format('ddd');
          return (
            <TableHeaderColumn
              key={dayNr}
              dataField={dayNr}
              csvHeader={dayStr}
              dataFormat={this.cellFormatter}
              columnClassName="charts_cell"
            >
              {dayStr}
            </TableHeaderColumn>
          );
        })}
        <TableHeaderColumn
          dataField="total"
          columnClassName="charts_cell"
          csvHeader="total"
        >
          Total
        </TableHeaderColumn>
      </TableWithoutOverflow>
    );
  }
}

export default ForecastChartsTable;
