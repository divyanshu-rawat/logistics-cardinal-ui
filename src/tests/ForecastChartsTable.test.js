import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ForecastChartsTable from '../ForecastChartsTable';
import { charts } from './dataBasedOnFixtures';

describe('<ForecastChartsTable />', () => {
  const sortedDates = Object.keys(charts[1].dates).sort();
  const renderedComponent = mount(
    <ForecastChartsTable
      chartData={{ 1: { sortedDates, data: charts[1].data } }}
      sortedDays={['1']}
    />,
  );

  it('should render the component', () => {
    const tree = toJson(renderedComponent);
    expect(tree).toMatchSnapshot();
  });
});
