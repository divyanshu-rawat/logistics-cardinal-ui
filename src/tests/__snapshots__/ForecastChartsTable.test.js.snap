// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastChartsTable /> should render the component 1`] = `
<ForecastChartsTable
  chartData={
    Object {
      "1": Object {
        "data": Array [
          Object {
            "2018-01-01": 11,
            "2018-01-08": 25,
            "formattedTime": "10:00:00",
          },
          Object {
            "2018-01-01": 50,
            "2018-01-08": 26,
            "formattedTime": "11:00:00",
          },
        ],
        "sortedDates": Array [
          "2018-01-01",
          "2018-01-08",
        ],
      },
    }
  }
  sortedDays={
    Array [
      "1",
    ]
  }
>
  <Styled(Styled(Table))
    data={
      Array [
        Object {
          "1": Object {
            "data": Array [
              Object {
                "2018-01-01": 11,
                "2018-01-08": 25,
                "formattedTime": "10:00:00",
              },
              Object {
                "2018-01-01": 50,
                "2018-01-08": 26,
                "formattedTime": "11:00:00",
              },
            ],
            "sortedDates": Array [
              "2018-01-01",
              "2018-01-08",
            ],
          },
        },
      ]
    }
    pagination={false}
    tableHeaderClass="hidden"
  >
    <Styled(Table)
      className="sc-bwzfXH gHROIr"
      data={
        Array [
          Object {
            "1": Object {
              "data": Array [
                Object {
                  "2018-01-01": 11,
                  "2018-01-08": 25,
                  "formattedTime": "10:00:00",
                },
                Object {
                  "2018-01-01": 50,
                  "2018-01-08": 26,
                  "formattedTime": "11:00:00",
                },
              ],
              "sortedDates": Array [
                "2018-01-01",
                "2018-01-08",
              ],
            },
          },
        ]
      }
      pagination={false}
      tableHeaderClass="hidden"
    >
      <Table
        className="sc-bwzfXH gHROIr sc-bdVaJa fxanBQ"
        data={
          Array [
            Object {
              "1": Object {
                "data": Array [
                  Object {
                    "2018-01-01": 11,
                    "2018-01-08": 25,
                    "formattedTime": "10:00:00",
                  },
                  Object {
                    "2018-01-01": 50,
                    "2018-01-08": 26,
                    "formattedTime": "11:00:00",
                  },
                ],
                "sortedDates": Array [
                  "2018-01-01",
                  "2018-01-08",
                ],
              },
            },
          ]
        }
        pagination={false}
        tableHeaderClass="hidden"
      >
        <BootstrapTable
          autoCollapse={
            Object {
              "filter": false,
              "search": false,
              "sort": false,
            }
          }
          bodyContainerClass={null}
          bordered={false}
          cellEdit={
            Object {
              "afterSaveCell": undefined,
              "beforeSaveCell": undefined,
              "blurToEscape": false,
              "blurToSave": false,
              "mode": "none",
              "nonEditableRows": undefined,
            }
          }
          className="sc-bwzfXH gHROIr sc-bdVaJa fxanBQ"
          columnFilter={false}
          condensed={false}
          containerClass={null}
          csvFileName="spreadsheet.csv"
          data={
            Array [
              Object {
                "1": Object {
                  "data": Array [
                    Object {
                      "2018-01-01": 11,
                      "2018-01-08": 25,
                      "formattedTime": "10:00:00",
                    },
                    Object {
                      "2018-01-01": 50,
                      "2018-01-08": 26,
                      "formattedTime": "11:00:00",
                    },
                  ],
                  "sortedDates": Array [
                    "2018-01-01",
                    "2018-01-08",
                  ],
                },
              },
            ]
          }
          deleteRow={false}
          expandColumnOptions={
            Object {
              "expandColumnBeforeSelectColumn": true,
              "expandColumnComponent": undefined,
              "expandColumnVisible": false,
            }
          }
          exportCSV={false}
          fetchInfo={
            Object {
              "dataTotalSize": 0,
            }
          }
          footer={false}
          headerContainerClass={null}
          height="100%"
          hover={false}
          ignoreSinglePage={false}
          insertRow={false}
          keyBoardNav={false}
          multiColumnSearch={false}
          multiColumnSort={1}
          options={
            Object {
              "afterColumnFilter": undefined,
              "afterDeleteRow": undefined,
              "afterInsertRow": undefined,
              "afterSearch": undefined,
              "afterTableComplete": undefined,
              "alwaysShowAllBtns": false,
              "beforeShowError": undefined,
              "btnGroup": undefined,
              "clearSearch": false,
              "clearSearchBtn": undefined,
              "closeText": "Close",
              "defaultSearch": "",
              "defaultSortName": undefined,
              "defaultSortOrder": undefined,
              "deleteBtn": undefined,
              "deleteText": "Delete",
              "excludeCSVHeader": false,
              "expandBodyClass": null,
              "expandBy": "row",
              "expandParentClass": null,
              "expandRowBgColor": undefined,
              "expanding": Array [],
              "exportCSVBtn": undefined,
              "exportCSVSeparator": ",",
              "exportCSVText": "Export to CSV",
              "firstPage": "<<",
              "firstPageTitle": "first page",
              "handleConfirmDeleteRow": undefined,
              "hidePageListOnlyOnePage": false,
              "hideSizePerPage": false,
              "ignoreEditable": false,
              "insertBtn": undefined,
              "insertFailIndicator": "Validation errors, please check!",
              "insertModal": undefined,
              "insertModalBody": undefined,
              "insertModalFooter": undefined,
              "insertModalHeader": undefined,
              "insertText": "New",
              "keepSizePerPageState": false,
              "lastPage": ">>",
              "lastPageTitle": "last page",
              "nextPage": ">",
              "nextPageTitle": "next page",
              "noAutoBOM": true,
              "noDataText": undefined,
              "onExpand": undefined,
              "onMouseEnter": undefined,
              "onMouseLeave": undefined,
              "onRowClick": undefined,
              "onRowDoubleClick": undefined,
              "onRowMouseOut": undefined,
              "onRowMouseOver": undefined,
              "onSizePerPageList": undefined,
              "onlyOneExpanding": false,
              "page": undefined,
              "pageStartIndex": 1,
              "paginationPanel": undefined,
              "paginationPosition": "bottom",
              "paginationShowsTotal": false,
              "paginationSize": 5,
              "prePage": "<",
              "prePageTitle": "previous page",
              "printToolBar": true,
              "saveText": "Save",
              "searchDelayTime": undefined,
              "searchField": undefined,
              "searchPanel": undefined,
              "searchPosition": "right",
              "showSelectedOnlyBtn": undefined,
              "sizePerPage": undefined,
              "sizePerPageDropDown": undefined,
              "sizePerPageList": Array [
                10,
                25,
                30,
                50,
              ],
              "sortIndicator": true,
              "sortName": undefined,
              "sortOrder": undefined,
              "toolBar": undefined,
              "toolbarPosition": "top",
              "withFirstAndLast": true,
              "withoutNoDataText": false,
            }
          }
          pagination={false}
          printable={false}
          replace={false}
          search={false}
          selectRow={
            Object {
              "bgColor": "",
              "clickToExpand": false,
              "clickToSelect": false,
              "clickToSelectAndEditCell": false,
              "customComponent": undefined,
              "hideSelectColumn": false,
              "mode": "none",
              "onSelect": undefined,
              "onSelectAll": undefined,
              "onlyUnselectVisible": false,
              "selected": Array [],
              "showOnlySelected": false,
              "unselectable": Array [],
            }
          }
          striped={true}
          tableBodyClass={null}
          tableContainerClass={null}
          tableFooterClass={null}
          tableHeaderClass="hidden"
          trClassName=""
          version="3"
          withoutTabIndex={false}
        >
          <div
            className="react-bs-table-container sc-bwzfXH gHROIr sc-bdVaJa fxanBQ"
          >
            <div
              className="react-bs-table"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "height": "100%",
                  "maxHeight": undefined,
                }
              }
            >
              <TableHeader
                bordered={false}
                colGroups={
                  <colgroup>
                    <col
                      style={
                        Object {
                          "display": null,
                          "minWidth": "100px",
                          "width": "100px",
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "display": null,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "display": null,
                        }
                      }
                    />
                  </colgroup>
                }
                condensed={false}
                expandColumnBeforeSelectColumn={true}
                expandColumnVisible={false}
                headerContainerClass={null}
                hideSelectColumn={false}
                isFiltered={false}
                isSelectAll={false}
                onSelectAllRow={[Function]}
                onSort={[Function]}
                reset={false}
                rowSelectType="none"
                sortIndicator={true}
                sortList={Array []}
                tableHeaderClass="hidden"
                version="3"
              >
                <div
                  className="react-bs-container-header table-header-wrapper"
                >
                  <table
                    className="table table-hover hidden"
                  >
                    <colgroup>
                      <col
                        key="0"
                        style={
                          Object {
                            "display": null,
                            "minWidth": "100px",
                            "width": "100px",
                          }
                        }
                      />
                      <col
                        key="1"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      />
                      <col
                        key="2"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      />
                    </colgroup>
                    <thead>
                      <tr
                        key="0"
                      >
                        <TableHeaderColumn
                          className=""
                          columnClassName="charts_cell"
                          columnTitle={false}
                          csvFieldType="string"
                          csvHeader="week"
                          dataAlign="left"
                          dataField="week"
                          dataSort={false}
                          defaultASC={false}
                          editColumnClassName=""
                          editable={true}
                          expandable={true}
                          filterFormatted={false}
                          headerTitle={true}
                          hidden={false}
                          hiddenOnInsert={false}
                          invalidEditColumnClassName=""
                          isKey={true}
                          isOnlyHead={false}
                          key="1"
                          keyValidator={false}
                          onSort={[Function]}
                          reset={false}
                          searchable={true}
                          sortIndicator={true}
                          version="3"
                          width="100px"
                        >
                          <th
                            className=""
                            data-is-only-head={false}
                            onClick={[Function]}
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title="Week"
                          >
                            Week
                            <div
                              onClick={[Function]}
                            />
                          </th>
                        </TableHeaderColumn>
                        <TableHeaderColumn
                          className=""
                          columnClassName="charts_cell"
                          columnTitle={false}
                          csvFieldType="string"
                          csvHeader="Mon"
                          dataAlign="left"
                          dataField="1"
                          dataFormat={[Function]}
                          dataSort={false}
                          defaultASC={false}
                          editColumnClassName=""
                          editable={true}
                          expandable={true}
                          filterFormatted={false}
                          headerTitle={true}
                          hidden={false}
                          hiddenOnInsert={false}
                          invalidEditColumnClassName=""
                          isKey={false}
                          isOnlyHead={false}
                          key="2"
                          keyValidator={false}
                          onSort={[Function]}
                          reset={false}
                          searchable={true}
                          sortIndicator={true}
                          version="3"
                          width={null}
                        >
                          <th
                            className=""
                            data-is-only-head={false}
                            onClick={[Function]}
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title="Mon"
                          >
                            Mon
                            <div
                              onClick={[Function]}
                            />
                          </th>
                        </TableHeaderColumn>
                        <TableHeaderColumn
                          className=""
                          columnClassName="charts_cell"
                          columnTitle={false}
                          csvFieldType="string"
                          csvHeader="total"
                          dataAlign="left"
                          dataField="total"
                          dataSort={false}
                          defaultASC={false}
                          editColumnClassName=""
                          editable={true}
                          expandable={true}
                          filterFormatted={false}
                          headerTitle={true}
                          hidden={false}
                          hiddenOnInsert={false}
                          invalidEditColumnClassName=""
                          isKey={false}
                          isOnlyHead={false}
                          key="3"
                          keyValidator={false}
                          onSort={[Function]}
                          reset={false}
                          searchable={true}
                          sortIndicator={true}
                          version="3"
                          width={null}
                        >
                          <th
                            className=""
                            data-is-only-head={false}
                            onClick={[Function]}
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title="Total"
                          >
                            Total
                            <div
                              onClick={[Function]}
                            />
                          </th>
                        </TableHeaderColumn>
                      </tr>
                    </thead>
                  </table>
                </div>
              </TableHeader>
              <TableBody
                bodyContainerClass={null}
                bordered={false}
                cellEdit={
                  Object {
                    "afterSaveCell": undefined,
                    "beforeSaveCell": undefined,
                    "blurToEscape": false,
                    "blurToSave": false,
                    "mode": "none",
                    "nonEditableRows": undefined,
                  }
                }
                columns={
                  Array [
                    Object {
                      "align": "left",
                      "attrs": undefined,
                      "className": "charts_cell",
                      "columnTitle": false,
                      "customEditor": undefined,
                      "editAttrs": undefined,
                      "editClassName": "",
                      "editable": true,
                      "expandable": true,
                      "export": undefined,
                      "filterFormatted": false,
                      "filterValue": undefined,
                      "format": undefined,
                      "formatExtraData": undefined,
                      "hidden": false,
                      "hiddenOnInsert": false,
                      "index": 0,
                      "invalidEditColumnClassName": "",
                      "name": "week",
                      "searchable": true,
                      "sort": false,
                      "sortFunc": undefined,
                      "sortFuncExtraData": undefined,
                      "style": undefined,
                      "text": "Week",
                      "width": "100px",
                    },
                    Object {
                      "align": "left",
                      "attrs": undefined,
                      "className": "charts_cell",
                      "columnTitle": false,
                      "customEditor": undefined,
                      "editAttrs": undefined,
                      "editClassName": "",
                      "editable": true,
                      "expandable": true,
                      "export": undefined,
                      "filterFormatted": false,
                      "filterValue": undefined,
                      "format": [Function],
                      "formatExtraData": undefined,
                      "hidden": false,
                      "hiddenOnInsert": false,
                      "index": 1,
                      "invalidEditColumnClassName": "",
                      "name": "1",
                      "searchable": true,
                      "sort": false,
                      "sortFunc": undefined,
                      "sortFuncExtraData": undefined,
                      "style": undefined,
                      "text": "Mon",
                      "width": null,
                    },
                    Object {
                      "align": "left",
                      "attrs": undefined,
                      "className": "charts_cell",
                      "columnTitle": false,
                      "customEditor": undefined,
                      "editAttrs": undefined,
                      "editClassName": "",
                      "editable": true,
                      "expandable": true,
                      "export": undefined,
                      "filterFormatted": false,
                      "filterValue": undefined,
                      "format": undefined,
                      "formatExtraData": undefined,
                      "hidden": false,
                      "hiddenOnInsert": false,
                      "index": 2,
                      "invalidEditColumnClassName": "",
                      "name": "total",
                      "searchable": true,
                      "sort": false,
                      "sortFunc": undefined,
                      "sortFuncExtraData": undefined,
                      "style": undefined,
                      "text": "Total",
                      "width": null,
                    },
                  ]
                }
                condensed={false}
                data={
                  Array [
                    Object {
                      "1": Object {
                        "data": Array [
                          Object {
                            "2018-01-01": 11,
                            "2018-01-08": 25,
                            "formattedTime": "10:00:00",
                          },
                          Object {
                            "2018-01-01": 50,
                            "2018-01-08": 26,
                            "formattedTime": "11:00:00",
                          },
                        ],
                        "sortedDates": Array [
                          "2018-01-01",
                          "2018-01-08",
                        ],
                      },
                    },
                  ]
                }
                expandBodyClass={null}
                expandBy="row"
                expandColumnOptions={
                  Object {
                    "expandColumnBeforeSelectColumn": true,
                    "expandColumnComponent": undefined,
                    "expandColumnVisible": false,
                  }
                }
                expandParentClass={null}
                expanding={Array []}
                hover={false}
                keyBoardNav={false}
                keyField="week"
                onEditCell={[Function]}
                onExpand={[Function]}
                onNavigateCell={[Function]}
                onRowClick={[Function]}
                onRowDoubleClick={[Function]}
                onRowMouseOut={[Function]}
                onRowMouseOver={[Function]}
                onSelectRow={[Function]}
                onlyOneExpanding={false}
                selectRow={
                  Object {
                    "bgColor": "",
                    "clickToExpand": false,
                    "clickToSelect": false,
                    "clickToSelectAndEditCell": false,
                    "customComponent": undefined,
                    "hideSelectColumn": false,
                    "mode": "none",
                    "onSelect": undefined,
                    "onSelectAll": undefined,
                    "onlyUnselectVisible": false,
                    "selected": Array [],
                    "showOnlySelected": false,
                    "unselectable": Array [],
                  }
                }
                selectedRowKeys={Array []}
                striped={true}
                style={
                  Object {
                    "height": "100%",
                    "maxHeight": undefined,
                  }
                }
                tableBodyClass={null}
                trClassName=""
                version="3"
                withoutNoDataText={false}
                withoutTabIndex={false}
                x={-1}
                y={-1}
              >
                <div
                  className="react-bs-container-body"
                  style={
                    Object {
                      "height": "100%",
                      "maxHeight": undefined,
                    }
                  }
                >
                  <table
                    className="table table-striped"
                  >
                    <colgroup>
                      <col
                        key="0"
                        style={
                          Object {
                            "display": null,
                            "minWidth": "100px",
                            "width": "100px",
                          }
                        }
                      />
                      <col
                        key="1"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      />
                      <col
                        key="2"
                        style={
                          Object {
                            "display": null,
                          }
                        }
                      />
                    </colgroup>
                    <tbody>
                      <TableRow
                        className=""
                        dbClickToEdit={false}
                        enableCellEdit={false}
                        index={0}
                        isSelected={false}
                        onExpandRow={[Function]}
                        onRowClick={[Function]}
                        onRowDoubleClick={[Function]}
                        onRowMouseOut={[Function]}
                        onRowMouseOver={[Function]}
                        onSelectRow={[Function]}
                        row={
                          Object {
                            "1": Object {
                              "data": Array [
                                Object {
                                  "2018-01-01": 11,
                                  "2018-01-08": 25,
                                  "formattedTime": "10:00:00",
                                },
                                Object {
                                  "2018-01-01": 50,
                                  "2018-01-08": 26,
                                  "formattedTime": "11:00:00",
                                },
                              ],
                              "sortedDates": Array [
                                "2018-01-01",
                                "2018-01-08",
                              ],
                            },
                          }
                        }
                        unselectableRow={false}
                      >
                        <tr
                          className=""
                          onClick={[Function]}
                          onDoubleClick={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                          style={Object {}}
                        >
                          <TableColumn
                            cIndex={0}
                            cellEdit={
                              Object {
                                "afterSaveCell": undefined,
                                "beforeSaveCell": undefined,
                                "blurToEscape": false,
                                "blurToSave": false,
                                "mode": "none",
                                "nonEditableRows": undefined,
                              }
                            }
                            className="charts_cell"
                            columnTitle={null}
                            customNavStyle={null}
                            dataAlign="left"
                            hidden={false}
                            isFocus={false}
                            key="0"
                            keyBoardNav={false}
                            onClick={[Function]}
                            onEdit={[Function]}
                            onKeyDown={[Function]}
                            rIndex={0}
                            row={
                              Object {
                                "1": Object {
                                  "data": Array [
                                    Object {
                                      "2018-01-01": 11,
                                      "2018-01-08": 25,
                                      "formattedTime": "10:00:00",
                                    },
                                    Object {
                                      "2018-01-01": 50,
                                      "2018-01-08": 26,
                                      "formattedTime": "11:00:00",
                                    },
                                  ],
                                  "sortedDates": Array [
                                    "2018-01-01",
                                    "2018-01-08",
                                  ],
                                },
                              }
                            }
                            tabIndex="1"
                            width="100px"
                            withoutTabIndex={false}
                          >
                            <td
                              className="charts_cell"
                              onClick={[Function]}
                              style={
                                Object {
                                  "display": null,
                                  "textAlign": "left",
                                }
                              }
                              tabIndex="1"
                              title={null}
                            />
                          </TableColumn>
                          <TableColumn
                            cIndex={1}
                            cellEdit={
                              Object {
                                "afterSaveCell": undefined,
                                "beforeSaveCell": undefined,
                                "blurToEscape": false,
                                "blurToSave": false,
                                "mode": "none",
                                "nonEditableRows": undefined,
                              }
                            }
                            className="charts_cell"
                            columnTitle={null}
                            customNavStyle={null}
                            dataAlign="left"
                            hidden={false}
                            isFocus={false}
                            key="1"
                            keyBoardNav={false}
                            onClick={[Function]}
                            onEdit={[Function]}
                            onKeyDown={[Function]}
                            rIndex={0}
                            row={
                              Object {
                                "1": Object {
                                  "data": Array [
                                    Object {
                                      "2018-01-01": 11,
                                      "2018-01-08": 25,
                                      "formattedTime": "10:00:00",
                                    },
                                    Object {
                                      "2018-01-01": 50,
                                      "2018-01-08": 26,
                                      "formattedTime": "11:00:00",
                                    },
                                  ],
                                  "sortedDates": Array [
                                    "2018-01-01",
                                    "2018-01-08",
                                  ],
                                },
                              }
                            }
                            tabIndex="2"
                            width={null}
                            withoutTabIndex={false}
                          >
                            <td
                              className="charts_cell"
                              onClick={[Function]}
                              style={
                                Object {
                                  "display": null,
                                  "textAlign": "left",
                                }
                              }
                              tabIndex="2"
                              title={null}
                            >
                              <div
                                className="chart-container"
                              >
                                0
                              </div>
                            </td>
                          </TableColumn>
                          <TableColumn
                            cIndex={2}
                            cellEdit={
                              Object {
                                "afterSaveCell": undefined,
                                "beforeSaveCell": undefined,
                                "blurToEscape": false,
                                "blurToSave": false,
                                "mode": "none",
                                "nonEditableRows": undefined,
                              }
                            }
                            className="charts_cell"
                            columnTitle={null}
                            customNavStyle={null}
                            dataAlign="left"
                            hidden={false}
                            isFocus={false}
                            key="2"
                            keyBoardNav={false}
                            onClick={[Function]}
                            onEdit={[Function]}
                            onKeyDown={[Function]}
                            rIndex={0}
                            row={
                              Object {
                                "1": Object {
                                  "data": Array [
                                    Object {
                                      "2018-01-01": 11,
                                      "2018-01-08": 25,
                                      "formattedTime": "10:00:00",
                                    },
                                    Object {
                                      "2018-01-01": 50,
                                      "2018-01-08": 26,
                                      "formattedTime": "11:00:00",
                                    },
                                  ],
                                  "sortedDates": Array [
                                    "2018-01-01",
                                    "2018-01-08",
                                  ],
                                },
                              }
                            }
                            tabIndex="3"
                            width={null}
                            withoutTabIndex={false}
                          >
                            <td
                              className="charts_cell"
                              onClick={[Function]}
                              style={
                                Object {
                                  "display": null,
                                  "textAlign": "left",
                                }
                              }
                              tabIndex="3"
                              title={null}
                            />
                          </TableColumn>
                        </tr>
                      </TableRow>
                    </tbody>
                  </table>
                </div>
              </TableBody>
            </div>
            <SAlert
              stack={
                Object {
                  "limit": 3,
                }
              }
            >
              <div
                className="s-alert-wrapper"
              />
            </SAlert>
          </div>
        </BootstrapTable>
      </Table>
    </Styled(Table)>
  </Styled(Styled(Table))>
</ForecastChartsTable>
`;
